container/pull.bzl

```diff
@@ -191,6 +191,14 @@ exports_files(["image.digest", "digest"])
         fail("Failed to read digest: %s" % digest_result.stderr)
     updated_attrs["digest"] = digest_result.stdout
 
+    if repository_ctx.attr.digest and repository_ctx.attr.digest != updated_attrs["digest"]:
+        fail(("SHA256 of the image specified does not match SHA256 of the pulled image. " +
+              "Expected {}, but pulled image with {}. " +
+              "It is possible that you pin to a manifest list which points to another image").format(
```

Comment:
It is possible that you have a pin to a ...
which points to another image, if so, ... <provide advice as to how to solve>